File: Ensembles_Arithmétique/Code/pgcd.rs
fn pgcd(n: usize, m: usize) -> usize {
    
    let mut a: usize;
    let mut b: usize;
    let mut c: usize;

    if n >= m {
        a = n;
        b = m
    } else {
        a = m;
        b = n;
    }

    while b>0 {
        c = b;
        b = a - (a/b)*b;
        a = c;
    }

    a
}

File: Ensembles_Arithmétique/Code/pgcd.c
int pgcd(unsigned int n, unsigned int m) {
    
    unsigned int a;
    unsigned int b;
    unsigned int c;

    if (n <= m) {
        a = m;
        b = n;
    } else {
        a = n;
        b = m;
    }
    
    while (b > 0) {
        c = a % b;
        a = b;
        b = c;
    }

    return a;
}
